app/src/main/java/lt/vilnius/tvarkau/utils/SharedPrefsManager.java

```diff
@@ -0,0 +1,77 @@
+package lt.vilnius.tvarkau.utils;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.util.Log;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+
+import lt.vilnius.tvarkau.entity.Profile;
+
+/**
+ * Managing user's shared Preferences.
+ * At the time it is only used to track if user has opted for anonymous login
+ * but later on it also will help us to store actual user information.
+ */
+public class SharedPrefsManager {
+    private static final String PREFS_NAME = "TVARKAU-VILNIU_PREFS";
+
+    private static final String PREF_ISANONYMOUS = "isUserAnonymous";
+    private static final Boolean DEFAULT_PREF_ISANONYMOUS = true;
+
+    private static final String PREF_USER_PROFILE = "UserProfile";
+    private static final String PREF_DEFAULT_USER_PROFILE = "[]";
+
+    private static SharedPrefsManager mSingleton = new SharedPrefsManager();
+    private static Context mContext;
+
+
+    public static SharedPrefsManager instance(Context context) {
+        mContext = context;
+        return mSingleton;
+    }
+
+    public static SharedPrefsManager instance() {
+        return mSingleton;
+    }
+
+    private static void init(Context context) {
+        mContext = context;
+    }
+
+    public SharedPreferences getPrefs() {
+        return mContext.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE);
+    }
+
+    public Boolean getIsUserAnonymous() {
+        return getPrefs().getBoolean(PREF_ISANONYMOUS, DEFAULT_PREF_ISANONYMOUS);
```

Comment:
Change it to `return null == getPrefs().getString(PREF_USER_PROFILE, null)`. Also remove both `PREF_ISANONYMOUS` and `DEFAULT_PREF_ISANONYMOUS`.